packages/framework/attributor/src/mixinAttributor.ts

```diff
@@ -16,92 +12,32 @@ import {
-} from "@fluidframework/telemetry-utils/internal";
-
-import { Attributor, type IAttributor, OpStreamAttributor } from "./attributor.js";
-import { AttributorSerializer, type Encoder, chain, deltaEncoder } from "./encoders.js";
-import { makeLZ4Encoder } from "./lz4Encoder.js";
-
-// Summary tree keys
-const attributorTreeName = ".attributor";
-const opBlobName = "op";
-
-/**
- * @internal
- */
-export const enableOnNewFileKey = "Fluid.Attribution.EnableOnNewFile";
-
-/**
- * @internal
- */
-export const IRuntimeAttributor: keyof IProvideRuntimeAttributor = "IRuntimeAttributor";
-
-/**
- * @internal
- */
-export interface IProvideRuntimeAttributor {
-	readonly IRuntimeAttributor: IRuntimeAttributor;
-}
-
-/**
- * Provides access to attribution information stored on the container runtime.
- *
- * @remarks Attributors are only populated after the container runtime into which they are being injected has initialized.
- *
- * @sealed
- * @internal
- */
-export interface IRuntimeAttributor extends IProvideRuntimeAttributor {
-	/**
-	 * @throws - If no AttributionInfo exists for this key.
-	 */
-	get(key: AttributionKey): AttributionInfo;
+import { assert } from "@fluidframework/core-utils/internal";
+import { type NamedFluidDataStoreRegistryEntries } from "@fluidframework/runtime-definitions/internal";
+import { loggerToMonitoringContext } from "@fluidframework/telemetry-utils/internal";
 
-	/**
-	 * @returns Whether any AttributionInfo exists for the provided key.
-	 */
-	has(key: AttributionKey): boolean;
-
-	/**
-	 * @returns Whether the runtime is currently tracking attribution information for the loaded container.
-	 * If enabled, the runtime attributor can be asked for the attribution info for different keys.
-	 * See {@link mixinAttributor} for more details on when this happens.
-	 */
-	readonly isEnabled: boolean;
-}
+import {
+	attributorDataStoreAlias,
+	enableOnNewFileKey,
+	type IRuntimeAttributor,
+} from "./attributorContracts.js";
+import type { RuntimeAttributorDataStoreChannel } from "./runtimeAttributorDataStoreChannel.js";
+import { RuntimeAttributorFactory } from "./runtimeAttributorDataStoreFactory.js";
 
 /**
- * Creates an `IRuntimeAttributor` for usage with {@link mixinAttributor}.
- *
- * @remarks The attributor will only be populated with data once it's passed via scope to a container runtime load flow.
- *
+ * Utility function to get the runtime attributor from the container runtime.
+ * @param runtime - container runtime from which attributor is to be fetched.
+ * @returns IRuntimeAttributor if it exists, otherwise undefined.
  * @internal
  */
-export function createRuntimeAttributor(): IRuntimeAttributor {
-	return new RuntimeAttributor();
+export async function getRuntimeAttributor(
+	runtime: IContainerRuntime,
+): Promise<IRuntimeAttributor | undefined> {
+	const entryPoint = await runtime.getAliasedDataStoreEntryPoint(attributorDataStoreAlias);
+	const runtimeAttributor = (await entryPoint?.get()) as
```

Comment:
Shouldn't this do `.IRuntimeAttributor` instead of the cast, since this thing uses the provider pattern?  cc @anthony-murphy - Is this a "legit" case for querying via provider pattern?

Reply:
changed